lib/schedule/schedule_page.dart

```diff
@@ -16,6 +15,18 @@ class SchedulePage extends StatefulWidget {
 
 class SchedulePageState extends State<SchedulePage>
     with SingleTickerProviderStateMixin {
+
+  ScheduleApi api = ScheduleApi();
+  List<Session> _sessions = List<Session>();
+  bool _isLoading = false;
+
+  @override
+  void initState() {
+    // TODO: implement initState
+    super.initState();
+    _fetchSessions();
```

Comment:
Instead of calling it here. We can have `_fetchSessions` return a `Future<List<Session>>` and use `FutureBuilder` inside build method as a top-most widget.